### ISSUE TITLE
ensure dataset tree api only returns nodes which contain leaves that are datasets

### ISSUE THREAD

**Issue by dpsutton** [metabase/metabase#18672]
when we return the datasets (similar to saved questions) we need to only return nodes which have actual datasets (or saved questions) below them. IE,

```
Root
└── collection
    ├── collection
    │   └── saved-question
    └── collection <- no saved-questions so don't return this node
```

Need to clear with product that this change can happen to saved questions as well. seems like an obvious win but is a change. Will be easier to achieve once the rest of the UI is in place so deferred to a later step
